pkg/apiserver/cmd/start.go

```diff
@@ -0,0 +1,309 @@
+	storageNamespace string,
+	apis []*builders.APIGroupBuilder,
+	openapidefs openapi.GetOpenAPIDefinitions,
+	title,
+	version string) {
+	logs.InitLogs()
+	defer logs.FlushLogs()
+
+	GetOpenApiDefinition = openapidefs
+
+	signalCh := genericapiserver.SetupSignalHandler()
+	// To disable providers, manually specify the list provided by getKnownProviders()
+	cmd := NewCommandStartServer(restConfig, storageNamespace, apis, signalCh, title, version)
+
+	cmd.Flags().AddFlagSet(pflag.CommandLine)
+	if err := cmd.Execute(); err != nil {
+		panic(err)
+	}
+}
+
+// NewCommandStartServer create a cobra start server command
+func NewCommandStartServer(restConfig *rest.Config, storageNamespace string, builders []*builders.APIGroupBuilder,
+	stopCh <-chan struct{}, title, version string) *cobra.Command {
+	o := NewServerOptions(restConfig, storageNamespace, builders)
+
+	// Support overrides
+	cmd := &cobra.Command{
+		Short: "Launch an API server",
+		Long:  "Launch an API server",
+		RunE: func(c *cobra.Command, args []string) error {
+			return o.RunServer(stopCh, title, version)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&o.PrintBearerToken, "print-bearer-token", false,
+		"Print a curl command with the bearer token to test the server")
+	flags.BoolVar(&o.RunDelegatedAuth, "delegated-auth", true,
+		"Setup delegated auth")
+	o.RecommendedOptions.AddFlags(flags)
+	o.InsecureServingOptions.AddFlags(flags)
+	feature.DefaultFeatureGate.AddFlag(flags)
+
+	klogFlags := flag.NewFlagSet("klog", flag.ExitOnError)
+	klog.InitFlags(klogFlags)
+	flags.AddGoFlagSet(klogFlags)
+
+	// Sync the glog and klog flags.
+	klogFlags.VisitAll(func(f *flag.Flag) {
+		goFlag := flag.CommandLine.Lookup(f.Name)
+		if goFlag != nil {
+			goFlag.Value.Set(f.Value.String())
+		}
+	})
+
+	return cmd
+}
+
+// NewServerOptions create a ServerOptions
+func NewServerOptions(restConfig *rest.Config, storageNamespace string, b []*builders.APIGroupBuilder) *ServerOptions {
+	versions := []schema.GroupVersion{}
+	for _, b := range b {
+		versions = append(versions, b.GetLegacyCodec()...)
+	}
+
+	codec := builders.Codecs.LegacyCodec(versions...)
+	o := &ServerOptions{
+		RecommendedOptions: genericoptions.NewRecommendedOptions("", codec),
+		APIBuilders:        b,
+		RunDelegatedAuth:   true,
+		RestConfig:         restConfig,
+		StorageNamespace:   storageNamespace,
+		Codec:              codec,
+	}
+	o.RecommendedOptions.SecureServing.BindPort = 443
+
+	o.RecommendedOptions.Authorization.RemoteKubeConfigFileOptional = true
+	o.RecommendedOptions.Authentication.RemoteKubeConfigFileOptional = true
+	o.InsecureServingOptions = func() *genericoptions.DeprecatedInsecureServingOptionsWithLoopback {
+		o := genericoptions.DeprecatedInsecureServingOptions{}
+		return o.WithLoopback()
+	}()
+
+	return o
+}
+
+// Config create an apiserver config
+func (o ServerOptions) Config() (*apiserver.Config, error) {
+	if err := o.RecommendedOptions.SecureServing.MaybeDefaultWithSelfSignedCerts(
```

Comment:
The CA issue is left for later resolution

Reply:
self-sign certs and skip tls verify should be enough for internal use